### PR TITLE
Implement tp_name and derive support

### PR DESCRIPTION
**Pull request by youknowone**
I finished `pystruct_sequence` work but not about pyclass.

### CODE REVIEW

File: derive/src/pyclass.rs [hunk 1]
```diff
@@ -420,6 +420,7 @@ pub fn impl_pyimpl(attr: AttributeArgs, item: Item) -> Result<TokenStream2, Diag
 fn generate_class_def(
     ident: &Ident,
     name: &str,
+    tp_name: &str,
```
Comment (coolreader18) [inline derive/src/pyclass.rs:423]: Is there a similar thing to this in Python? A tp_name slot or something? If not, I'd recommend calling this `QUALNAME` or similar.

File: vm/src/pyobject.rs [hunk 2]
```diff
@@ -1407,6 +1407,7 @@ where
 
 pub trait PyClassDef {
     const NAME: &'static str;
+    const TP_NAME: &'static str;
```
Comment (coolreader18) [inline vm/src/pyobject.rs:1410]:
```suggestion
    const TP_NAME: &'static str = Self::NAME;
```

File: derive/src/pyclass.rs (hunk 1 above)
Comment (youknowone) [reply derive/src/pyclass.rs:423]: Yes, there is `tp_name`. Here is a test code about it:
```c
.tp_name = "custom.Custom",
```

File: vm/src/pyobject.rs (hunk 2 above)
Comment (youknowone) [reply vm/src/pyobject.rs:1410]: I don't think this default value is good. `tp_name` in CPython includes module name.

Comment (coolreader18) [reply vm/src/pyobject.rs:1410]: Alright, this looks fine to me then.

**Pull request merged**